proposals/improved-interpolated-strings.md

```diff
@@ -291,6 +291,39 @@ This was done to support partial formatting scenarios where the user wants to st
 introduce a bunch of unnecessary branches in standard interpolated string usage. We could consider an addendum where we use just `Format` methods if no `TryFormat` method is present, but
 it does present questions about what we do if there's a mix of both TryFormat and Format calls.
 
+### Passing previous arguments to the builder
+
+There is unfortunate lack of symmetry in the proposal at it currently exists: invoking an extension method in reduced form produces different semantics than invoking the extension method in
+normal form. This is different from most other locations in the language, where reduced form is just a sugar. We have a couple of potential options for resolving this:
+
+* Special case extension methods called in normal form. This feels pretty icky: why are extensions special here?
+* Allow other previous parameters to be passed to the builder. This gets complicated quickly: how do we determine what to pass to the builder? What if the builder has a `GetInterpolatedString`
+method that accepts the first parameter, but not the receiver, of an instance method?
+* Pass parameters to the builder marked with a specific attribute, a la `EnumeratorCancellation` support. This would need rules about whether we pass the receiver (maybe if the method is marked
+we pass the receiver, and we don't in the general case?), and what we do if parameters _after_ the string parameter are annotated, but it seems like a potential option.
+
+Some compromise is likely needed here, but either direction has complications. Some scenarios that would be affected by this is the `Utf8Formatter` below, or existing api patterns that have
+an `IFormatProvider` as the first argument.
+
+### `await` usage in interpolation holes
+
+Because `$"{await A()}"` is a valid expression today, we need to rationalize how interpolation holes with await. We could solve this with a few rules:
+
+1. If an interpolated string used as a `string`, `IFormattable`, or `FormattableString` has an `await` in an interpolation hole, fall back to old-style formatter.
+2. If an interpolated string is subject to an _implicit\_string\_builder\_conversion_ and _applicable\_interpolated\_string\_builder\_type_ is a `ref struct`, `await` is not allowed to be used
+in the format holes.
+
+Fundamentally, this desugaring could use a ref struct in an async method as long as we guarantee that the `ref struct` will not need to be saved to the heap, which should be possible if we forbid
+`await`s in the interpolation holes.
+
+Alternatively, we could simply make all builder types non-ref structs, including the framework builder for interpolated strings. This would, however, preclude us from someday recognizing a `Span`
+version that does not need to allocate any scratch space at all.
```

Comment:
I think we should say:
- InterpolatedStringBuilder is a ref struct.
- If you're targeting strings, the compiler choose between its various ways of implementing string interpolation: InterpolatedStringBuilder, string.Concat, and string.Format.  If there's something that prevents it from using InterpolatedStringBuilder, well then it uses one of its other mechanisms.
- If you're targeting a custom builder, then you're subject to the constraints of that builder.  If the builder is a struct or class, great, you can use await in holes; have at it.  If the builder is a ref struct, then you can't use await in holes, just as you can't use anything the builder doesn't have a TryFormat overload for.